> This notebook can be used to generate mutant sequences

In [2]:
import pandas as pd
import numpy as np 
import os

from helpers import*

### 1. Single Mutant sequences

In [ ]:
wt_seq = 'MLDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVSDPRLADNPLIAINQAFTDLTGYSEEECVGRNCRFLAGSGTEPWLTDKIRQGVREHKPVLVEILNYKKDGTPFRNAVLVAPIYDDDDELLYFLGSQVEVDDDQPNMGMARRERAAEMLKTLSPRQLEVTTLVASGLRNKEVAARLGLSEKTVKMHRGLVMEKLNLKTSADLVRIAVEAGI'

In [ ]:
mutations = generate_mutations(wt_seq) # Generate single mutants
 
# Save the generated dataframe: 
    
df = pd.DataFrame(mutations)
df = df[['Position', 'Original_AA', 'Mutated_Position', 'Mutated_AA', 'seq']]

df.to_csv('generated/single_mutants.csv', index=False)

In [3]:
df.head()

Position Original_AA  Mutated_Position Mutated_AA  \
0         1           M                 1          A   
1         1           M                 1          C   
2         1           M                 1          D   
3         1           M                 1          E   
4         1           M                 1          F   

                                                 seq  
0  ALDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
1  CLDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
2  DLDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
3  ELDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
4  FLDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...

##### Remarks:

In [4]:
print(19*len(wt_seq))

4275


> There are 4275 possible sequences for single mutant proteins.

In [5]:
double_mutants = generate_double_mutants(wt_seq) # Generate double mutants
    
# Save the generating file:
    
df2 = pd.DataFrame(double_mutants)
df2 = df2[['Position1', 'Original_AA1', 'Mutated_Position1', 'Mutated_AA1', 'Position2', 'Original_AA2', 'Mutated_Position2', 'Mutated_AA2', 'seq']]
df2.to_csv('generated/double_mutants.csv', index=False)

df2.head()


Position1 Original_AA1  Mutated_Position1 Mutated_AA1  Position2  \
0          1            M                  1           A          2   
1          1            M                  1           A          2   
2          1            M                  1           A          2   
3          1            M                  1           A          2   
4          1            M                  1           A          2   

  Original_AA2  Mutated_Position2 Mutated_AA2  \
0            L                  2           A   
1            L                  2           C   
2            L                  2           D   
3            L                  2           E   
4            L                  2           F   

                                                 seq  
0  AADMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
1  ACDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
2  ADDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
3  AEDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...  
4  AFDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVS...

##### Remarks

In [6]:
print(df2.shape[0])

(9097200, 9)

In [7]:
# This is the theorical number of double mutants

number_of_double_mutants = 19**2 * (len_wt_seq * (len_wt_seq - 1)) // 2
print("Number of double mutants (excluding original amino acids):", number_of_double_mutants)

Number of double mutants (excluding original amino acids): 9097200


In [3]:
#split the double_mutants.csv in different file because it is too large to train otherwise.

input_csv_path = 'generated/double_mutants.csv'
output_folder = 'data'
chunk_size = 1000000  # Adjust the chunk size as needed

split_csv(input_csv_path, output_folder, chunk_size)


> 